lib/src/header_parser/sub_parsers/macro_parser.dart

```diff
@@ -131,17 +132,41 @@ int _macroVariablevisitor(Pointer<clang_types.CXCursor> cursor,
           );
           break;
         case clang_types.CXEvalResultKind.CXEval_StrLiteral:
-          var value = Utf8.fromUtf8(clang.clang_EvalResult_getAsStr(e).cast());
-          // Escape $ character.
-          value = value.replaceAll(r'$', r'\$');
-          // Escape ' character, because our strings are enclosed with '.
-          value = value.replaceAll("'", r"\'");
+          final strPtr = clang.clang_EvalResult_getAsStr(e); // Do not free.
+          final length = Utf8.strlen(strPtr.cast());
+          final charList = Uint8List.view(
+              strPtr.cast<Uint8>().asTypedList(length).buffer, 0, length);
+
+          final sb = StringBuffer();
+          for (final char in charList) {
+            if (char >= 32 && char <= 126) {
+              // Handle the printable range of ASCII.
+              final printableChar = String.fromCharCode(char);
+              // Escape $ and ' character.
+              if (printableChar == '\$' ||
+                  printableChar == "'" ||
+                  printableChar == '\\') {
+                sb.write('\\$printableChar');
```

Comment:
You could possibly use raw strings instead to prevent having to escape the chars: `r'This will end with a newline.\n'`

Reply:
Yes I suppose we can, this will handle `$` and `\` but we'd still need to escape `'`.

Reply:
> Yes I suppose we can

We can't actually, because we may have to write things like `\x0A` to our string and also escape `'`.